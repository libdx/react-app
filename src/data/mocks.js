//@flow

import type { Movie } from '../movie'

export const forrest: Movie = {
    imageURL: "http://t0.gstatic.com/images?q=tbn:ANd9GcSppDgk99BKVA4TJtWc1FN4-HUkdWrFNfMm1-M0nQ01sIOcbTZu",
    title: "Forrest Gump",
    releaseYear: "1994",
    genre: "Comedy-drama",
    description: "Slow-witted Forrest Gump (Tom Hanks) has never thought of himself as disadvantaged, and thanks to his supportive mother (Sally Field), he leads anything but a restricted life. Whether dominating on the gridiron as a college football star, fighting in Vietnam or captaining a shrimp boat, Forrest inspires people with his childlike optimism. But one person Forrest cares about most may be the most difficult to save -- his childhood love, the sweet but troubled Jenny (Robin Wright)."
}

export const movies = [forrest, forrest, forrest, forrest, forrest, forrest]

