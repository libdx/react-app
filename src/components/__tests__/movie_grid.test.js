import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import MovieGrid  from '../movie_grid'
import MovieCard from '../movie_card'

// FIXME: fire out how to properly mock model objects
import { movies } from '../../data/mocks'

describe('MovieGrid', () => {

    let element
    let component

    beforeEach(() => {
        element = <MovieGrid movies={movies} />
        component = shallow(element)
    })

    it('has list of movies', () => {
        const list = component.find('MovieCard') 
        expect(list.length).toBe(movies.length)
    })

    it('renders correctly', () => {
        const tree = renderer.create(element).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
