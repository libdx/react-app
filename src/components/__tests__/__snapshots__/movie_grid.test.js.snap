// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieGrid renders correctly 1`] = `
<div
  className="row"
>
  <div
    className="col-md-4 col-sm-6"
  >
    <div
      className="card movie-card"
    >
      <img
        className="card-img-top poster-img"
        src="http://t0.gstatic.com/images?q=tbn:ANd9GcSppDgk99BKVA4TJtWc1FN4-HUkdWrFNfMm1-M0nQ01sIOcbTZu"
      />
      <div
        className="card-body movie-card-body"
      >
        <h6
          className="card-title textbox"
        >
          <span
            className="alignleft"
          >
            Forrest Gump
          </span>
          <span
            className="badge badge-light alignright"
          >
            1994
          </span>
          <div
            className="alighclear"
          />
        </h6>
        <p
          className="card-subtitle mb-2 text-muted"
        >
          Comedy-drama
        </p>
      </div>
    </div>
  </div>
  <div
    className="col-md-4 col-sm-6"
  >
    <div
      className="card movie-card"
    >
      <img
        className="card-img-top poster-img"
        src="http://t0.gstatic.com/images?q=tbn:ANd9GcSppDgk99BKVA4TJtWc1FN4-HUkdWrFNfMm1-M0nQ01sIOcbTZu"
      />
      <div
        className="card-body movie-card-body"
      >
        <h6
          className="card-title textbox"
        >
          <span
            className="alignleft"
          >
            Forrest Gump
          </span>
          <span
            className="badge badge-light alignright"
          >
            1994
          </span>
          <div
            className="alighclear"
          />
        </h6>
        <p
          className="card-subtitle mb-2 text-muted"
        >
          Comedy-drama
        </p>
      </div>
    </div>
  </div>
  <div
    className="col-md-4 col-sm-6"
  >
    <div
      className="card movie-card"
    >
      <img
        className="card-img-top poster-img"
        src="http://t0.gstatic.com/images?q=tbn:ANd9GcSppDgk99BKVA4TJtWc1FN4-HUkdWrFNfMm1-M0nQ01sIOcbTZu"
      />
      <div
        className="card-body movie-card-body"
      >
        <h6
          className="card-title textbox"
        >
          <span
            className="alignleft"
          >
            Forrest Gump
          </span>
          <span
            className="badge badge-light alignright"
          >
            1994
          </span>
          <div
            className="alighclear"
          />
        </h6>
        <p
          className="card-subtitle mb-2 text-muted"
        >
          Comedy-drama
        </p>
      </div>
    </div>
  </div>
  <div
    className="col-md-4 col-sm-6"
  >
    <div
      className="card movie-card"
    >
      <img
        className="card-img-top poster-img"
        src="http://t0.gstatic.com/images?q=tbn:ANd9GcSppDgk99BKVA4TJtWc1FN4-HUkdWrFNfMm1-M0nQ01sIOcbTZu"
      />
      <div
        className="card-body movie-card-body"
      >
        <h6
          className="card-title textbox"
        >
          <span
            className="alignleft"
          >
            Forrest Gump
          </span>
          <span
            className="badge badge-light alignright"
          >
            1994
          </span>
          <div
            className="alighclear"
          />
        </h6>
        <p
          className="card-subtitle mb-2 text-muted"
        >
          Comedy-drama
        </p>
      </div>
    </div>
  </div>
  <div
    className="col-md-4 col-sm-6"
  >
    <div
      className="card movie-card"
    >
      <img
        className="card-img-top poster-img"
        src="http://t0.gstatic.com/images?q=tbn:ANd9GcSppDgk99BKVA4TJtWc1FN4-HUkdWrFNfMm1-M0nQ01sIOcbTZu"
      />
      <div
        className="card-body movie-card-body"
      >
        <h6
          className="card-title textbox"
        >
          <span
            className="alignleft"
          >
            Forrest Gump
          </span>
          <span
            className="badge badge-light alignright"
          >
            1994
          </span>
          <div
            className="alighclear"
          />
        </h6>
        <p
          className="card-subtitle mb-2 text-muted"
        >
          Comedy-drama
        </p>
      </div>
    </div>
  </div>
  <div
    className="col-md-4 col-sm-6"
  >
    <div
      className="card movie-card"
    >
      <img
        className="card-img-top poster-img"
        src="http://t0.gstatic.com/images?q=tbn:ANd9GcSppDgk99BKVA4TJtWc1FN4-HUkdWrFNfMm1-M0nQ01sIOcbTZu"
      />
      <div
        className="card-body movie-card-body"
      >
        <h6
          className="card-title textbox"
        >
          <span
            className="alignleft"
          >
            Forrest Gump
          </span>
          <span
            className="badge badge-light alignright"
          >
            1994
          </span>
          <div
            className="alighclear"
          />
        </h6>
        <p
          className="card-subtitle mb-2 text-muted"
        >
          Comedy-drama
        </p>
      </div>
    </div>
  </div>
</div>
`;
